wolfcrypt/src/asn.c

```diff
@@ -3460,6 +3460,69 @@ word32 SetBitString(word32 len, byte unusedBits, byte* output)
 #endif /* !NO_RSA || HAVE_ECC || HAVE_ED25519 || HAVE_ED448 */
 
 #ifdef ASN_BER_TO_DER
+
+#define BER_OCTET_LENGTH 4096
+
+/* sets the terminating 0x00 0x00 at the end of an indefinite length
+ * returns the number of bytes written */
+word32 SetIndefEnd(byte* in)
+{
+    byte terminate[] = { 0x00, 0x00 };
+
+    if (in != NULL) {
+        XMEMCPY(in, terminate, 2);
```

Comment:
better to use `sizeof(terminate)` rather than literal `2`.

it would also be good to incorporate `ASN_INDEF_END_SZ` into the implementation, to guard against inconsistency with  `sizeof(terminate)`.  we don't have static asserts in the wolfssl repo, which is the right way to do this, but you can declare `byte terminate[ASN_INDEF_END_SZ] = ...` and we will catch any missing or extra initializer bytes in some of our tests.